gdx/test/com/badlogic/gdx/utils/QueueTest.java

```diff
@@ -374,6 +402,233 @@ public void hashEqualsTest () {
+		assertEquals(2, queue.values.length);
+		queue.ensureCapacity(2);
+		assertEquals(2, queue.values.length);
+	}
+
+	@Test
+	public void ensureCapacityNotFit () {
+		// Create the queue.
+		Queue<String> queue = new Queue<>(2, String.class);
+
+		// Add an element.
+		queue.addFirst("first");
+
+		// Enough space, does not change.
+		assertEquals(2, queue.values.length);
+		queue.ensureCapacity(2);
+		assertEquals(3, queue.values.length);
+	}
+
+	@Test
+	public void isEmpty () {
+		// Create the queue
+		Queue<String> queue = new Queue<>();
+
+		assertTrue(queue.isEmpty());
+		queue.addFirst("element");
+		assertFalse(queue.isEmpty());
+	}
+
+	@Test
+	public void notEmpty () {
+		// Create the queue
+		Queue<String> queue = new Queue<>();
+
+		assertFalse(queue.notEmpty());
+		queue.addFirst("element");
+		assertTrue(queue.notEmpty());
+	}
+
+	@Test
+	public void equalsIdentitySameObject () {
+		Queue<String> queue = new Queue<>();
+		assertTrue(queue.equalsIdentity(queue));
+	}
+
+	@Test
+	public void equalsIdentityDifferentObject () {
+		Queue<String> queue = new Queue<>();
+
+		assertFalse(queue.equalsIdentity(null));
+		assertFalse(queue.equalsIdentity("different"));
+	}
+
+	@Test
+	public void equalsIdentityDifferentQueueSameObjects () {
+		// Initialize queues.
+		Queue<String> queue1 = new Queue<>();
+		Queue<String> queue2 = new Queue<>();
+
+		// Check empty equality
+		assertTrue(queue1.equalsIdentity(queue2));
+		assertTrue(queue2.equalsIdentity(queue1));
+
+		// Check inequality.
+		queue1.addLast("1");
+		assertFalse(queue1.equalsIdentity(queue2));
+		assertFalse(queue2.equalsIdentity(queue1));
+
+		// Check equality.
+		queue2.addLast("1");
+		assertTrue(queue1.equalsIdentity(queue2));
+		assertTrue(queue2.equalsIdentity(queue1));
+	}
+
+	@Test
+	public void initializeQueueIterator () {
+		// Initialize queue and iterator.
+		Queue<String> queue = new Queue<>();
+		queue.addLast("remove");
+		Queue.QueueIterator<String> iterator = new Queue.QueueIterator<>(queue);
+
+		// Allow remove is set to true, thus test a remove case.
+		assertEquals(1, queue.size);
+		iterator.next();
+		iterator.remove();
+		assertEquals(0, queue.size);
+	}
+
+	@Test
+	public void queueIteratorInvalidHasNext () {
+		// Initialize queue and iterator.
+		Queue<Integer> queue = new Queue<>();
+		Queue.QueueIterator<Integer> iterator = new Queue.QueueIterator<>(queue);
+		iterator.valid = false;
+
+		// GdxRuntimeException will be thrown by check hasNext of invalid iterator.
+		exception.expect(GdxRuntimeException.class);
+		boolean hasNext = iterator.hasNext();
+	}
+
+	@Test
+	public void queueIteratorInvalidNext () {
+		// Initialize queue and iterator.
+		Queue<String> queue = new Queue<>();
+		Queue.QueueIterator<String> iterator = new Queue.QueueIterator<>(queue);
+		iterator.valid = false;
+		queue.addLast("1");
+
+		// GdxRuntimeException will be thrown by check hasNext of invalid iterator.
+		exception.expect(GdxRuntimeException.class);
```

Comment:
you shouldn't test undocumented exceptions, only test methods contract from its javadoc.
So i think you can remove this test.